app/SharedWebComponents/Pages/VoiceChat.razor.cs

```diff
@@ -79,28 +79,11 @@ private void OnSendPrompt()
                                 Name = voice
                             };
                         }
-                        try
-                        {
-                            SpeechSynthesis.Speak(utterance, duration =>
-                            {
-                                _isReadingResponse = false;
-                                StateHasChanged();
-                            });
-                        }
-                        catch
-                        {
-                            // TODO: Find a better way to do this
-                            // The Blazor.SpeechSynthesis.WebAssembly API supports the callback,
-                            // however the underlying code does not do this using a DI-friendly way.
-                            // The code assumes the concrete implementation for the ISpeechSynthesisService
-                            // service is the concrete Web Assembly type which is not valid. However, the
-                            // alternate API is something MAUI can implement.
 
-                            SpeechSynthesis.Speak(utterance);
+                        SpeechSynthesis.Speak(utterance);
 
-                            _isReadingResponse = false;
-                            StateHasChanged();
-                        }
+                        _isReadingResponse = false;
+                        StateHasChanged();
```

Comment:
Should this rather be after an await or some API? This looks to immediately start the speech and then say it is done. Before this was in a callback/delegate. Maybe MAUI did it wrong and use the async void so as not to change the code?

Reply:
Oh, you're right. The callback provided the `duration`. I think this should be reverted.